# MINI PROJETO - ANÁLISE DE DADOS COM PYTHON | MÓDULO 1
**Felipe Busarello**  
**Curso:** Análise de Dados com Python  
**Turma:** 05 - 2026  
**Professor:** Cláudio Neves

## ACESSAR O ARQUIVO BASE DE DADOS
* Acesse o link https://www.kaggle.com/datasets/namespaiva/base-varejo/data
* Faça o download do arquivo base-varejo.zip
* Instalar as bibliotecas Pandas e Numpy, auxiliares ao Python

## INSTALAÇÕES DAS BIBLIOTECAS UTILIZADAS

In [1]:
%pip install -q pandas # MANIPULAÇÃO E ANÁLISE DE DADOS
%pip install -q numpy # CÁLCULOS E OPERAÇÕES NUMÉRICAS
%pip install -q matplotlib # CRIAÇÃO DE GRÁFICOS
%pip install -q seaborn # VISUALIZAÇÕES ESTATÍSTICAS

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip
ERROR: Invalid requirement: '#': Expected package name at the start of dependency specifier
    #
    ^


Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip
ERROR: Invalid requirement: '#': Expected package name at the start of dependency specifier
    #
    ^


Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip
ERROR: Invalid requirement: '#': Expected package name at the start of dependency specifier
    #
    ^


Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip
ERROR: Invalid requirement: '#': Expected package name at the start of dependency specifier
    #
    ^


## IMPORTAÇÃO DAS BIBLIOTECAS UTILIZADAS

In [2]:
import pandas as pd
import os
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import csv
from IPython.display import display


# ===========
# VALIDAÇÃO DAS IMPORTAÇÕES E VERSÕES INSTALADAS
# ===========
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)
print("Matplotlib:", plt.matplotlib.__version__)
print("Seaborn:", sns.__version__)

Pandas: 3.0.5
NumPy: 2.5.1
Matplotlib: 3.11.1
Seaborn: 0.13.2


## LEITURA ARQUIVO BASE

In [3]:
with open("dataset/Base Varejo.csv", encoding="utf-8") as arquivo:
    leitor = csv.DictReader(arquivo)
    colunas = leitor.fieldnames
    vendas = list(leitor)


# ===========
# CONFERÊNCIA DA LEITURA
# ===========
print("VALIDAÇÃO LEITURA ARQUIVO .CSV\n")
print(f"Arquivo lido com sucesso!")
print(f"Quantidade de registros: {len(vendas)}")
print(f"Quantidade de colunas: {len(colunas)}")
print("Colunas:", colunas)
print("\nPrimeiro registro:")
print(vendas[0])

VALIDAÇÃO LEITURA ARQUIVO .CSV

Arquivo lido com sucesso!
Quantidade de registros: 830000
Quantidade de colunas: 1
Colunas: ['DATA;CO_ID;CL_ID;CL_GENERO;CL_EC;CL_FHL;CL_SEG;PR_ID;PR_CAT;PR_NOME;;;;']

Primeiro registro:
{'DATA;CO_ID;CL_ID;CL_GENERO;CL_EC;CL_FHL;CL_SEG;PR_ID;PR_CAT;PR_NOME;;;;': '01/02/2019;1000;534;M;4;1;C;67;BEBIDAS;REFRIGERANTE GUARANA;;;;'}


## Primeira validação
Com a primeira validação, é possível verificar que o `DictReader` está lendo o arquivo, porém está entendendo que a linha inteira é uma única coluna:
```
print(f"Quantidade de colunas: {len(colunas)}")
Quantidade de colunas: 1
```
Isso ocorre, pois o CSV está separando as informações com `;` e não `,` (O padrão de leitura do código `DictReader` é `,`).  
Dessa forma, é necessário informar ao `DictReader` que o limitador dessa base é `;`.

In [4]:
with open("dataset/Base Varejo.csv", encoding="utf-8") as arquivo:
    leitor = csv.DictReader(arquivo, delimiter=";") # INCLUIDO DELIMITADOR ";"
    colunas = leitor.fieldnames
    vendas = list(leitor)


# ===========
# NOVA CONFERÊNCIA DA LEITURA
# ===========
print("VALIDAÇÃO LEITURA ARQUIVO .CSV\n")
print(f"Arquivo lido com sucesso!")
print(f"Quantidade de registros: {len(vendas)}")
print(f"Quantidade de colunas: {len(colunas)}")
print("Colunas:", colunas)
print("\nPrimeiro registro:")
print(vendas[0])

VALIDAÇÃO LEITURA ARQUIVO .CSV

Arquivo lido com sucesso!
Quantidade de registros: 830000
Quantidade de colunas: 14
Colunas: ['DATA', 'CO_ID', 'CL_ID', 'CL_GENERO', 'CL_EC', 'CL_FHL', 'CL_SEG', 'PR_ID', 'PR_CAT', 'PR_NOME', '', '', '', '']

Primeiro registro:
{'DATA': '01/02/2019', 'CO_ID': '1000', 'CL_ID': '534', 'CL_GENERO': 'M', 'CL_EC': '4', 'CL_FHL': '1', 'CL_SEG': 'C', 'PR_ID': '67', 'PR_CAT': 'BEBIDAS', 'PR_NOME': 'REFRIGERANTE GUARANA', '': ''}


## LEITURA REALIZADA CORRETAMENTE
```
print(f"Quantidade de colunas: {len(colunas)}")
Quantidade de colunas: 14
```

## LEITURA BASE COM A BIBLIOTECA PANDAS

In [5]:
# ===========
# REALIZANDO A LEITURA UTILIZANDO PANDAS
# ===========
df_original = pd.read_csv("dataset/Base Varejo.csv", encoding="utf-8", delimiter=";") # COLOCANDO O DELIMITADOR ';' 
display(df_original.head()) # MOSTRANDO AS 5 PRIMEIRAS LINHAS DO DATAFRAME CRIADO

,DATA,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG,PR_ID,PR_CAT,PR_NOME,Unnamed: 10,Unnamed: 11,Unnamed: 12,Unnamed: 13
0,01/02/2019,1000,534,M,4,1,C,67,BEBIDAS,REFRIGERANTE GUARANA,NaN,NaN,NaN,NaN
1,01/02/2019,1000,534,M,4,1,C,70,BEBIDAS,REFRIGERANTE OUTROS,NaN,NaN,NaN,NaN
2,01/02/2019,1000,534,M,4,1,C,178,HIGIENE,LENCO UMEDECIDO,NaN,NaN,NaN,NaN
3,01/02/2019,1000,534,M,4,1,C,4,ALIMENTOS,ABACAXI,NaN,NaN,NaN,NaN
4,01/02/2019,1000,534,M,4,1,C,175,LIMPEZA,LIMPADOR MULTIUSO,NaN,NaN,NaN,NaN


## VALIDAÇÕES INICIAS

In [6]:
print("VISUALIZAÇÕES INICIAIS\n")

# .shape() MOSTRA O NÚMERO DE LINHAS E COLUNAS
print(f"Quantidade de linhas: {df_original.shape[0]}") # MOSTRA O NÚMERO DE LINHAS
print(f"Quantidade de colunas: {df_original.shape[1]}") # MOSTRA O NÚMERO DE COLUNAS

print("\nColunas:")
print(df_original.columns.tolist()) # CRIANDO UMA LISTA COM OS NOMES DA COLUNAS

print("\nPrimeiros 5 registros:")
display(df_original.head())

print("\nÚltimos 5 registros:")
display(df_original.tail())

VISUALIZAÇÕES INICIAIS

Quantidade de linhas: 830000
Quantidade de colunas: 14

Colunas:
['DATA', 'CO_ID', 'CL_ID', 'CL_GENERO', 'CL_EC', 'CL_FHL', 'CL_SEG', 'PR_ID', 'PR_CAT', 'PR_NOME', 'Unnamed: 10', 'Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13']

Primeiros 5 registros:


,DATA,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG,PR_ID,PR_CAT,PR_NOME,Unnamed: 10,Unnamed: 11,Unnamed: 12,Unnamed: 13
0,01/02/2019,1000,534,M,4,1,C,67,BEBIDAS,REFRIGERANTE GUARANA,NaN,NaN,NaN,NaN
1,01/02/2019,1000,534,M,4,1,C,70,BEBIDAS,REFRIGERANTE OUTROS,NaN,NaN,NaN,NaN
2,01/02/2019,1000,534,M,4,1,C,178,HIGIENE,LENCO UMEDECIDO,NaN,NaN,NaN,NaN
3,01/02/2019,1000,534,M,4,1,C,4,ALIMENTOS,ABACAXI,NaN,NaN,NaN,NaN
4,01/02/2019,1000,534,M,4,1,C,175,LIMPEZA,LIMPADOR MULTIUSO,NaN,NaN,NaN,NaN



Últimos 5 registros:


,DATA,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG,PR_ID,PR_CAT,PR_NOME,Unnamed: 10,Unnamed: 11,Unnamed: 12,Unnamed: 13
829995,19/08/2022,919822,155,F,2,0,B,183,ALIMENTOS,KETCHUP,NaN,NaN,NaN,NaN
829996,19/08/2022,919822,155,F,2,0,B,56,ALIMENTOS,QUEIJO MUSSARELA,NaN,NaN,NaN,NaN
829997,19/08/2022,919822,155,F,2,0,B,227,ALIMENTOS,ARROZ,NaN,NaN,NaN,NaN
829998,19/08/2022,919822,155,F,2,0,B,214,ALIMENTOS,CEBOLA,NaN,NaN,NaN,NaN
829999,19/08/2022,919822,155,F,2,0,B,59,ALIMENTOS,SALGADINHO,NaN,NaN,NaN,NaN


Nesse momento foi verifcado que o arquivo base possui 4 (quatro) colunas sem títulos:  
`Unnamed: 10` `Unnamed: 11` `Unnamed: 12` `Unnamed: 13`  
Necessário realizar novas validações para verificar possiveis registros.

## TRATAMENTO COLUNAS SEM TÍTULO

In [7]:
# ===========
# CRIADO O df_columns PARA AS COLUNAS SEM TÍTULO
# ===========
df_columns = df_original[["Unnamed: 10", "Unnamed: 11", "Unnamed: 12", "Unnamed: 13"]]

display(df_columns)

,Unnamed: 10,Unnamed: 11,Unnamed: 12,Unnamed: 13
0,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN
2,NaN,NaN,NaN,NaN
3,NaN,NaN,NaN,NaN
4,NaN,NaN,NaN,NaN
...,...,...,...,...
829995,NaN,NaN,NaN,NaN
829996,NaN,NaN,NaN,NaN
829997,NaN,NaN,NaN,NaN
829998,NaN,NaN,NaN,NaN


In [8]:
# ===========
# VERIFICAÇÃO df_columns
# ===========
df_columns.info()

<class 'pandas.DataFrame'>
RangeIndex: 830000 entries, 0 to 829999
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Unnamed: 10  0 non-null      float64
 1   Unnamed: 11  0 non-null      float64
 2   Unnamed: 12  0 non-null      float64
 3   Unnamed: 13  0 non-null      float64
dtypes: float64(4)
memory usage: 25.3 MB


As colunas não possuem valores preenchidos (`0 non-null`) e, portanto, não apresentam informações úteis para a análise. Elas serão removidas do `df_copia`, preservando o `df_original` para referência.

## CRIANDO df_copia

In [9]:
df_copia = df_original.copy()

# ===========
# REMOVENDO AS COLUNAS SEM INFORMAÇÕES
# ===========
df_copia = df_copia.dropna(axis=1, how="all")

print("Qtd. Colunas antes da limpeza:", df_original.shape)
print("Qtd. Colunas após a limpeza:", df_copia.shape)
print("\nColunas:")
print(df_copia.columns.tolist()) 

Qtd. Colunas antes da limpeza: (830000, 14)
Qtd. Colunas após a limpeza: (830000, 10)

Colunas:
['DATA', 'CO_ID', 'CL_ID', 'CL_GENERO', 'CL_EC', 'CL_FHL', 'CL_SEG', 'PR_ID', 'PR_CAT', 'PR_NOME']


## VALIDAÇÕES DO df_copia

In [10]:
# ===========
# CRIANDO VALIDAÇÕES DO df_copia
# ===========
df_copia.info() # VERIFICANDO AS INFORMAÇÕES DE NULOS E TIPO DE CADA COLUNA
display(df_copia.isnull().sum()) # VALIDANDO O TOTAL DE NULOS DE CADA COLUNA

<class 'pandas.DataFrame'>
RangeIndex: 830000 entries, 0 to 829999
Data columns (total 10 columns):
 #   Column     Non-Null Count   Dtype
---  ------     --------------   -----
 0   DATA       830000 non-null  str  
 1   CO_ID      830000 non-null  int64
 2   CL_ID      830000 non-null  int64
 3   CL_GENERO  830000 non-null  str  
 4   CL_EC      830000 non-null  int64
 5   CL_FHL     830000 non-null  int64
 6   CL_SEG     830000 non-null  str  
 7   PR_ID      830000 non-null  int64
 8   PR_CAT     830000 non-null  str  
 9   PR_NOME    830000 non-null  str  
dtypes: int64(5), str(5)
memory usage: 63.3 MB


DATA         0
CO_ID        0
CL_ID        0
CL_GENERO    0
CL_EC        0
CL_FHL       0
CL_SEG       0
PR_ID        0
PR_CAT       0
PR_NOME      0
dtype: int64

Após a remoção das colunas totalmente vazias (`Unnamed: 10` a `Unnamed: 13`), a base passou a conter 830.000 registros e 10 colunas, **sem valores nulos**.

## INFORMAÇÕES INCORRETAS NAS COLUNAS

In [11]:
# ===========
# VERIFICANDO TODAS COLUNAS COM INFORMAÇÕES INCORRETAS
# ===========

for coluna in df_copia.columns:
    nulos = df_copia[coluna].isnull().sum()
    espacos = df_copia[coluna].astype(str).str.strip().eq("").sum()
    nd = (df_copia[coluna] == "#N/D").sum()
    
    # VAMOS MOSTRAR APENAS AS COLUNAS QUE POSSUEM ALGUM VALOR INCORETO
    if nulos > 0 or espacos > 0 or nd > 0:
        print(f"\nCOLUNA: {coluna}")

        if nulos > 0:
            print(f"Valores nulos: {nulos}")

        if espacos > 0:
            print(f"Valores apenas com espaços: {espacos}")

        if nd > 0:
            print(f"Valores '#N/D': {nd}")



COLUNA: PR_CAT
Valores '#N/D': 3650

COLUNA: PR_NOME
Valores '#N/D': 3650


As colunas `PR_CAT` e `PR_NOME` apresentam informações preenchidas como `#N/D`. Dessa forma vamos realizar os seguintes tratamentos:  
`PR_CAT` = `#N/D` será substituido por `Sem Categoria`  
`PR_NOME` = `#N/D` será substituido por `Sem Nome`

In [12]:
# ===========
# TRATAMENTO DOS VALORES #N/D
# ===========
df_copia["PR_CAT"] = df_copia["PR_CAT"].replace("#N/D", "Sem Categoria")
df_copia["PR_NOME"] = df_copia["PR_NOME"].replace("#N/D", "Sem Nome")

# ===========
# VALIDAÇÃO DO TRATAMENTO
# ===========
print("PR_CAT - quantidade de #N/D:",
      (df_copia["PR_CAT"] == "#N/D").sum())

print("PR_NOME - quantidade de #N/D:",
      (df_copia["PR_NOME"] == "#N/D").sum())

print("\nPR_CAT - quantidade de 'Sem Categoria':",
      (df_copia["PR_CAT"] == "Sem Categoria").sum())

print("PR_NOME - quantidade de 'Sem Nome':",
      (df_copia["PR_NOME"] == "Sem Nome").sum())

PR_CAT - quantidade de #N/D: 0
PR_NOME - quantidade de #N/D: 0

PR_CAT - quantidade de 'Sem Categoria': 3650
PR_NOME - quantidade de 'Sem Nome': 3650


## DUPLICIDADE

In [13]:
# ===========
# VALIDANDO AS INFORMAÇÕES EM DUPLICIDADE
# ===========
df_copia.duplicated().sum()    # SOMANDO AS LINHAS QUE ESTÃO EM DUPLICIDADE

np.int64(96553)

Foram identificado 96553 linhas em duplicidade.  
Entendendo que, na base não deve ter duplicidade, foram realizado as retirados das linhas duplicadas.

In [14]:
# ===========
# REMOVENDO AS LINHAS EM DUPLICIDADE
# ===========
df_copia = df_copia.drop_duplicates()
print("Qtd. Linhas Duplicadas:", df_copia.duplicated().sum())
print("Qtd. Linhas antes da limpeza:", df_original.shape[0])
print("Qtd. Linhas após a limpeza:", df_copia.shape[0])

Qtd. Linhas Duplicadas: 0
Qtd. Linhas antes da limpeza: 830000
Qtd. Linhas após a limpeza: 733447


Após a remoção das colunas vazias e linhas duplicadas, foi realizado uma análise para verificar o `dtype` de cada coluna e realizado os ajustes necessários.

## AJUSTE TIPO DA COLUNA

In [15]:
# ===========
# VERIFICANDO O TIPO DE CADA COLUNA
# ===========
display(df_copia.dtypes)


DATA           str
CO_ID        int64
CL_ID        int64
CL_GENERO      str
CL_EC        int64
CL_FHL       int64
CL_SEG         str
PR_ID        int64
PR_CAT         str
PR_NOME        str
dtype: object

Verificando que as colunas `CO_ID`, `CL_ID` e `PR_ID` são códigos de indentificação de compras, clientes e produtos, não devem ser considerados números (`int64`) e sim textos (`str`).  
A coluna `DATA` está como texto (`str`), vamos ajustar para o tipo correto `datetime`.  

Necessário realizar as alterações para o `Dtype` corretos.

In [24]:
# ===========
# AJUSTANDO OS TIPOS DAS COLUNAS PARA O FORMATO CORRETO
# ===========
df_copia["CO_ID"] = df_copia["CO_ID"].astype("str")
df_copia["CL_ID"] = df_copia["CL_ID"].astype("str")
df_copia["PR_ID"] = df_copia["PR_ID"].astype("str")

df_copia["DATA"] = pd.to_datetime(
    df_copia["DATA"],
    format="%d/%m/%Y",
    errors="coerce"
)

# ===========
# VALIDANDO AS ALTERAÇÕES REALIZADAS
# ===========
display(df_original[["DATA","CO_ID","CL_ID","PR_ID"]].dtypes)
display(df_copia[["DATA","CO_ID","CL_ID","PR_ID"]].dtypes)



DATA       str
CO_ID    int64
CL_ID    int64
PR_ID    int64
dtype: object

DATA     datetime64[us]
CO_ID               str
CL_ID               str
PR_ID               str
dtype: object

In [33]:
# ===========
# VALIDANDO A COLUNA DATA E POSSÍVEIS DATAS INVÁLIDAS (FORA DO PADRÃO)
# ===========
data_invalidas = df_copia["DATA"].isna().sum()
print(f"Quantidade de datas inválidas: {data_invalidas}")

if data_invalidas == 0:
    print("Não foram identificadas datas inválidas durante a conversão.")
    
# ===========
# VALIDANDO O PERÍODO DAS DATAS
# ===========
print("\nData mais antiga:", df_copia["DATA"].min())
print("Data mais recente:", df_copia["DATA"].max())

Quantidade de datas inválidas: 0
Não foram identificadas datas inválidas durante a conversão.

Data mais antiga: 2019-01-04 00:00:00
Data mais recente: 2022-12-08 00:00:00


## ANÁLISES ESTATÍSTIVAS DESCRITIVAS
SERÁ REALIZADO ANÁLISES INICIAIS NA COLUNA `CL_FHL`

In [42]:
# ===========
# VERIFICANDO AS ESTATÍTICAS BÁSICAS NA COLUNA `CL_FHL` 
# ===========
df_copia["CL_FHL"].describe()

count    733447.000000
mean          1.146049
std           1.416917
min           0.000000
25%           0.000000
50%           0.000000
75%           2.000000
max           4.000000
Name: CL_FHL, dtype: float64

A FUNÇÃO `.describe()` mostra os seguintes resultados:  

| ESTATÍSTICA | RESULTADO | INTERPRETAÇÃO |
| --------- | --------- | --------- |
| count | 733.447 | EXISTEM 733.447 REGISTROS COM INFORMAÇÃO NA COLUNA `CL_FHL` |
| mean | 1,146 | EM MÉDIA, OS REGISTROS POSSUEM APROXIMADAMTENTE 1,15 FILHOS |
| std/ Desvio Padrão | 1,417 | EXISTE UM DESVIO CONSIDERAVEL EM RELAÇÃO A MÉDIA |
| min | 0 | EXISTEM REGISTROS DE CLIENTES SEM FILHOS |
| 1º Quartil (25%) | 0 | PELO MENOS 25% DOS REGISTROS NÃO POSSUEM FILHOS |
| Mediana (50%) | 0 | METADO DOS REGISTROS NÃO POSSUEM FILHOS |
| 3º Quartil (75%) | 2 | 75% DOS REGISTROS POSSUEM ATÉ 2 FILHOS |
| max | 4 | O MAIOR NÚMERO DE FILHOS É 4 |

ALÉM DAS INFORMAÇÕES DO `.describe()` TAMBÉM VAMOS REALIZAR O CALCULO DA MODA PARA VERIFICAR O VALOR QUE APARECE COM MAIOR FREQUENCIA NA COLUNA `CL_FHL`

In [43]:
# ===========
# CÁLCULANDO A MODA
# ===========
moda = df_copia["CL_FHL"].mode()

print("Moda:", moda.iloc[0])

Moda: 0


ASSIM COMO A FUNÇÃO `.describe()` MOSTROU, A MEDIANA (50%) DOS REGISTROS NÃO POSSUEM FILHOS (0), A MODA CONFIRMOU ISSO.

In [50]:
# ===========
# AVALIANDO A DISTRIBUIÇÃO DE FILHOS
# ===========
distribuicao_filhos = (df_copia["CL_FHL"].value_counts().sort_index().to_frame(name="Quantidade"))
distribuicao_filhos["Percentual"] = (distribuicao_filhos["Quantidade"] / distribuicao_filhos["Quantidade"].sum()* 100)
distribuicao_filhos["Percentual"] = (distribuicao_filhos["Percentual"].map(lambda x: f"{x:.2f}%"))

display(distribuicao_filhos)

,Quantidade,Percentual
CL_FHL,,
0,384986,52.49%
1,90845,12.39%
2,94168,12.84%
3,92407,12.60%
4,71041,9.69%


## INSIGHT:
1. ASSIM COMO A `MODA` E `MEDIANA`, A DISTRIBUIÇÃO EM PERCENTUAL MOSTRO QUE `52,49%` DOS REGISTROS NÃO POSSUEM FILHOS;
2. RETIRANDO OS CLIENTES SEM FILHO (52,49%), A DISTRIBUIÇÃO DOS CLIENTES COM 1 ATÉ 4 FILHOS ESTÁ EQUILIBRADA, MANTENDO QUASE IGUAL NOS CLIENTES COM 1 ATÉ 3 FILHOS;
3. POSSÍVEL VERIFICAR QUE OS CLIENTES SEM FILHO, COM 1 E 2 FILHOS REPRESENTAM QUASE 78%, OU SEJA 3º QUARTIL APRESENTADO ANTERIORMENTE.  
  
OBS: COMO O df_copia NÃO INFORMA SE CADA REGISTRO REPRESENTA 1 CLIENTE, OU SE O CLIENTE PODE REPETIR VÁRIAS VEZES, É NECESSÁRIO FAZER ESSA VERIFICAÇÃO PARA GARANTIR QUE OS INSIGHT A CIMA SÃO POR CLIENTE OU POR VENDA

In [58]:
# ===========
# AVALIANDO DE CLIENTES ÚNICOS
# ===========
clientes_unicos = df_copia["CL_ID"].nunique()

print(f"Total de registro: {len(df_copia)}")
print(f"Total de clientes únicos: {clientes_unicos}")
if clientes_unicos != len(df_copia):
    print("Existem clientes com mais de um registro na base.")

Total de registro: 733447
Total de clientes únicos: 1000
Existem clientes com mais de um registro na base.


In [57]:
# ===========
# CONTABILIZANDO A FREQUENCIA POR `CL_ID`
# ===========
frequencia_clientes = df_copia["CL_ID"].value_counts()
display(frequencia_clientes.head(5))

CL_ID
822    1532
485    1433
397    1342
369    1334
971    1315
Name: count, dtype: int64

CONFORME INFORMADO ANTERIORMENTE, A BASE MOSTRA A FREQUÊNCIA DE COMPRA DOS CLIENTES, DESSA FORMA O CLIENTE PODE REPETIR DIVERSAS VEZES.  
EXEMPLO O CLIENTE `CL_ID` 822 QUE POSSUI UMA FREQUENCIA DE 1.532 COMPRAS.